File: 백준/Silver/1406. 에디터/에디터.cc
#include <iostream>
#include <stack>
#include <string>
using namespace std;

int main() {
	string str;
	cin >> str;

	stack<char> left;
	stack<char> right;

	for (int i = 0; i < str.size(); i++) {
		left.push(str[i]);
	}

	int cnt = 0;
	cin >> cnt;

	while (cnt--) {
		char input;
		cin >> input;

		if (input == 'L' && !left.empty()) {
			right.push(left.top());
			left.pop();
		}
		else if (input == 'D' && !right.empty()) {
			left.push(right.top());
			right.pop();
		}
		else if (input == 'B' && !left.empty()) {
			left.pop();
		}
		else if (input == 'P') {
			cin >> input;
			left.push(input);
		}
	}

	while (!left.empty()) {
		right.push(left.top());
		left.pop();
	}
	while (!right.empty()) {
		cout << right.top();
		right.pop();
	}

	return 0;
}

File: 백준/Silver/1406. 에디터/에디터.java
import java.io.*;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Stack<Character> left = new Stack<>();
        Stack<Character> right = new Stack<>();

        String input = br.readLine();

        for (char c : input.toCharArray()) {
            left.push(c);
        }

        int num = Integer.parseInt(br.readLine());

        while (num-- > 0) {
            String command = br.readLine();
            if (command.equals("L") && !left.isEmpty()) {
                right.push(left.pop());
            } else if (command.equals("D") && !right.isEmpty()) {
                left.push(right.pop());
            } else if (command.equals("B") && !left.isEmpty()) {
                left.pop();
            } else if (command.charAt(0) == 'P') {
                left.push(command.charAt(2));
            }
        }

        while (!left.isEmpty()) {
            right.push(left.pop());
        }

        while (!right.isEmpty()) {
            sb.append(right.pop());
        }

        System.out.println(sb);
    }
}